Book-A-Desk.Api/SmtpClientManager.fs

```diff
@@ -0,0 +1,20 @@
+namespace Book_A_Desk.Api
+
+open System.Net.Mail
+
+type SmtpClientManager =
+    {
+        SmtpClient: SmtpClient
+    }
+
+module SmtpClientManager =
+    let provide getEmailServiceConfiguration =         
+        let getSmtpClient =
+            let config = getEmailServiceConfiguration()
```

Comment:
I think there's an issue here.
In configureApp in Program.fs, you do:
`let getEmailServiceConfiguration = (fun () -> app.ApplicationServices.GetService<EmailServiceConfiguration>())`

To delay fetching the configuration until later, but then immediately pass it to SmtpClientManager.provide getEmailServiceConfiguration, which will then call getEmailServiceConfiguration immediately.

If you want to delay fetching the configuration, getSmtpClient should be a function instead of a value.

Reply:
It doesn't make sense after my last refactor. It has been fixed.